Makefile

```diff
@@ -232,6 +232,7 @@ CFLAGS += -ffunction-sections -fdata-sections -fno-strict-aliasing
 CFLAGS += -fno-builtin -fshort-enums
 CFLAGS += -D__HEAP_SIZE=1024
 CFLAGS += -D__STACK_SIZE=4096
+CFLAGS += -std=gnu23

```

Comment:
Why was this change needed?

Reply:
Default standard setup in the compiler did not have `static_assert` defined, probably was selecting a std before C11. Explicitly selected C23 as it's the latest supported std

Reply:
Thanks for the clarification, do you think using the new standard might cause any bugs? I have no idea on forwards compatibly between C++ standards.

Reply:
In general the C standard is very conservative. Forward compatibility is essentially a certainty unless one uses "reserved" identifiers or "exquisite" voodoo which is replaced by functionality in newer standards. 

No C23 (nor C17) features are being used in the code as of now, technically C11 would be enough.

No reserved identifiers that collide with functionality included up to C23 are in use either,  and no vodoo that is exposed to C standard updates is being used.

In summary, updating the standard used to compile should not affect the functionality of the firmware apart from providing the "new" functionality

Reply:
Great, thanks for explaining. Please, go ahead, mark this as resolved and merge the PR.